FWCore/Modules/src/LogErrorHarvester.cc

```diff
@@ -41,10 +46,42 @@ namespace edm {
     void beginJob() override;
     void produce(StreamID, Event&, EventSetup const&) const override;
     void endJob() override;
+    
   };
 
-  LogErrorHarvester::LogErrorHarvester(ParameterSet const&) {
+  LogErrorHarvester::LogErrorHarvester(ParameterSet const& iPSet) {
+    
     produces<std::vector<ErrorSummaryEntry>>();
+
+    const edm::TypeID endPathStatusType{typeid(edm::EndPathStatus)};
+    const edm::TypeID pathStatusType{typeid(edm::PathStatus)};
+    const edm::TypeID triggerResultsType{typeid(edm::TriggerResults)};
+
+    auto const& ignore = iPSet.getUntrackedParameter<std::vector<std::string>>("excludeModules");
+    const std::unordered_set<std::string> excludedModules(ignore.begin(),ignore.end());
+
+    auto const& includeM = iPSet.getUntrackedParameter<std::vector<std::string>>("includeModules");
+    const std::unordered_set<std::string> includeModules(includeM.begin(),includeM.end());
+
+    //Need to be sure to run only after all other EDProducers have run
+    callWhenNewProductsRegistered([this,
+                                   endPathStatusType,pathStatusType,triggerResultsType,
+                                   excludedModules, includeModules](edm::BranchDescription const& iBD) 
+    {
+      if((iBD.branchType() == edm::InEvent and moduleDescription().processName() == iBD.processName()) and 
+         ( (includeModules.empty() or
+            includeModules.end() != includeModules.find(iBD.moduleLabel())) and
+           (iBD.unwrappedTypeID() != endPathStatusType and
+            iBD.unwrappedTypeID() != pathStatusType and
+            iBD.unwrappedTypeID() != triggerResultsType))) {
+        if(excludedModules.end() == excludedModules.find(iBD.moduleLabel())) {
+          consumes(edm::TypeToGet{iBD.unwrappedTypeID(),edm::PRODUCT_TYPE},
+                   edm::InputTag{iBD.moduleLabel(),
+                                 iBD.productInstanceName(),
+                                 iBD.processName()});
```

Comment:
does this interfere with the earlyDelete?
If it does, we will need to look for another solution.

Reply:
IIUC, everything in the earlyDelete goes to the excludedModules, which means the logErrorHarvester will not wait for these.
So, if the producer of products from earlyDelete happens to run after all of the modules in includeModules,
the error harvesting for it will be missing.

Reply:
Theoretically correct. However, data products that are marked as 'earlyDelete' are made by modules doing 'intermediate' work. As long as the logErrorHarvester is waiting on something form a module that consumes the 'intermediate' we will get the messages.

Reply:
Right. OK then.
In my test jobs I see  84 excludeModules and  676 in includeModules.